bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/VehicleHandler.java

```diff
@@ -0,0 +1,576 @@
+                        guessRangeRadius(rangeFuel.get()), 0);
+                updateChannel(radiusFuel);
+                int hybridKm = rangeElectric.get().intValue() + rangeFuel.get().intValue();
+                QuantityType<Length> hybridRangeState = QuantityType.valueOf(hybridKm, KILOMETRE_UNIT);
+                ChannelStateMap rangeHybrid = new ChannelStateMap("range-hybrid", GROUP_RANGE, hybridRangeState, 0);
+                updateChannel(rangeHybrid);
+                ChannelStateMap radiusHybrid = new ChannelStateMap("radius-hybrid", GROUP_RANGE,
+                        guessRangeRadius(hybridRangeState), 0);
+                updateChannel(radiusHybrid);
+            }
+        } else if (rangeFuel.isPresent()) {
+            // update fuel & hybrid radius
+            ChannelStateMap radiusFuel = new ChannelStateMap("radius-fuel", GROUP_RANGE,
+                    guessRangeRadius(rangeFuel.get()), 0);
+            updateChannel(radiusFuel);
+        }
+    }
+
+    /**
+     * Easy function but there's some measures behind:
+     * Guessing the range of the Vehicle on Map. If you can drive x kilometers with your Vehicle it's not feasible to
+     * project this x km Radius on Map. The roads to be taken are causing some overhead because they are not a straight
+     * line from Location A to B.
+     * I've taken some measurements to calculate the overhead factor based on Google Maps
+     * Berlin - Dresden: Road Distance: 193 air-line Distance 167 = Factor 87%
+     * Kassel - Frankfurt: Road Distance: 199 air-line Distance 143 = Factor 72%
+     * After measuring more distances you'll find out that the outcome is between 70% and 90%. So
+     *
+     * This depends also on the roads of a concrete route but this is only a guess without any Route Navigation behind
+     *
+     * @param range
+     * @return mapping from air-line distance to "real road" distance
+     */
+    public static State guessRangeRadius(QuantityType<?> s) {
+        double radius = s.intValue() * 0.8;
+        return QuantityType.valueOf(Math.round(radius), KILOMETRE_UNIT);
+    }
+
+    protected void updateChannel(ChannelStateMap csm) {
+        updateTime(csm.getGroup(), csm.getTimestamp());
+        updateState(new ChannelUID(thing.getUID(), csm.getGroup(), csm.getChannel()), csm.getState());
+    }
+
+    private void updateTime(String group, long timestamp) {
+        boolean updateTime = false;
+        Long l = timeHash.get(group);
+        if (l != null) {
+            if (l.longValue() < timestamp) {
+                updateTime = true;
+            }
+        } else {
+            updateTime = true;
+        }
+        if (updateTime) {
+            timeHash.put(group, timestamp);
+            Date d = new Date(timestamp);
+            LocalDateTime ld = d.toInstant().atZone(Constants.zoneId).toLocalDateTime();
+            DateTimeType dtt = DateTimeType.valueOf(ld.format(DATE_INPUT_PATTERN));
```

Comment:
This seems overly complex and can be achieved without `Date` and `LocalDateTime`:
```suggestion
            DateTimeType dtt = new DateTimeType(ZonedDateTime.ofInstant(Instant.ofEpochMilli(timestamp), Constants.zoneId));
```

Reply:
Done, I took the example 2 from [DateTime conversion tutorial](https://community.openhab.org/t/datetime-conversion-openhab-3-x/107197)